app/controllers/carto/api/visualizations_controller.rb

```diff
@@ -281,13 +281,18 @@ def list
         tags = params.fetch(:tags, '').split(',')
         tags = nil if tags.empty?
         is_common_data_user = common_data_user && user_id == common_data_user.id
+        is_maps = types.count == 1 && types[0] == "'derived'"
```

Comment:
Minor:

Maybe you could just check the whole thing:
```ruby
is_maps = types == ["'derived'"]
```

If you do that then you can remove the extra variable and just do this below:

```ruby
if types == ["'derived'"]
  user_join = 'LEFT JOIN users AS u ON u.id=v.user_id'
  auth_tokens_column = ', ARRAY[u.auth_token] AS auth_tokens'
end
```